docs/docs/features/backlight.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Backlight
+sidebar_label: Backlight
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Backlight is a feature used to control array of LEDs, usually placed through or under switches. Unlike [RGB Underglow](underglow.md), backlight currently allows only one color per LED, also LEDs are not addressable, so you can't control individual LEDs.
+
+## Enabling Backlight
+
+To enable backlight on your board or shield, add the following line to your `.conf` file of your user config directory as such:
+
+```
+CONFIG_ZMK_BACKLIGHT=y
+```
+
+If your board or shield does not have backlight configured, refer to [Adding Backlight to a board or a shield](#adding-backlight-to-a-board-or-a-shield).
+
+## Configuring Backlight
+
+There are various Kconfig options used to configure the backlight feature. These can all be set in the `.conf` file.
+
+| Option                               | Description                                           | Default |
+| ------------------------------------ | ----------------------------------------------------- | ------- |
+| `CONFIG_ZMK_BACKLIGHT_BRT_STEP`      | Brightness step in percent                            | 20      |
+| `CONFIG_ZMK_BACKLIGHT_BRT_START`     | Default brightness in percent                         | 40      |
+| `CONFIG_ZMK_BACKLIGHT_ON_START`      | Default backlight state                               | y       |
+| `CONFIG_ZMK_BACKLIGHT_AUTO_OFF_IDLE` | Turn off backlight when keyboard goes into idle state | n       |
+| `CONFIG_ZMK_BACKLIGHT_AUTO_OFF_USB`  | Turn off backlight when USB is disconnected           | n       |
+
+## Adding Backlight to a board or a shield
+
+<Tabs
+defaultValue="shield"
+values={[
+{label: 'Adding to a board', value: 'board'},
+{label: 'Adding to a shield', value: 'shield'},
+]}>
+<TabItem value="board">
+
+First, you must enable PWM by adding the following lines to your `.conf` file:
```

Comment:
For this and the shield, I think we might want this in the `Kconfig` files so we can enable these *only* it the backlight config is enabled.

Thoughts?

Reply:
Not sure where it should be added, inside the `Kconfig.defconfig` file?
And should it be added to a `board/Kconfig.<board>` file for shields?

Reply:
For boards, this should definitely go in the `Kconfig.defconfig` file.

For shields, according to https://docs.zephyrproject.org/2.5.0/guides/porting/shields.html#board-specific-shield-configuration this would go into `<board>/<board>.defconfig`

Reply:
We put these in board_defconfig for our boards

Reply:
That works if you don't want users the option to disable a certain feature. But if a user might want to disable backlight, and not pull in other drivers unnecessarily, then the Kconfig.defaults approach actually allows that nuance/dependency chain.

Reply:
Fixed